aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -82,14 +88,33 @@ void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
         if (nv.isNone()) {
           continue;
         }
-        auto* interpreter = nv.unsafeToTensorImpl()->pyobj_slot()->pyobj_interpreter();
-        if (interpreter) {
-          (*interpreter)->dispatch(op, stack);
-          return;
+
+        auto* t = nv.unsafeToTensorImpl();
+        if (t->key_set().has(c10::DispatchKey::Python)) {
+          tensors_with_python_key_present = true;
+        }
+
+        auto* t_interpreter = t->pyobj_slot()->pyobj_interpreter();
+        if (!interpreter && t_interpreter) {
+          interpreter = t_interpreter;
         }
       }
     }
   }
+
+  if (interpreter) {
+    if (tensors_with_python_key_present) {
+      (*interpreter)->dispatch(op, stack);
+    } else {
+      // At this point, there are no modes in the stack and no tensors with the python key.
+      // so remove and disable the python key before redispatching.
```

Comment:
Could be useful to add a link to the issue here just to point out when this might happen